### PR TITLE
feat: add crypto explanation for the stream gateway

### PR DESCRIPTION
**Pull request by morphis**

### CODE REVIEW

File: explanation/cryptography/crypto_stream_gateway.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+Anbox Streaming Gateway is using cryptographic technology for:
+
+* TLS transport encryption
+* Mutual TLS based authentication
+* Token based authentication
+
+## TLS transport encryption
+
+All network endpoints exposed by the Anbox Stream Gateway are secured with TLS using an 4096 bit RSA key. The Anbox Stream Gateway strictly enforces TLS 1.3 or better and does not provide backward compatibility with older TLS versions.
```
Comment (keirthana) [inline explanation/cryptography/crypto_stream_gateway.md:12]:
```suggestion
All network endpoints exposed by the Anbox Stream Gateway are secured with TLS using an 4096 bit RSA key. The Anbox Stream Gateway strictly enforces TLS 1.3 or later and does not provide backward compatibility with older TLS versions.
```

File: explanation/cryptography/crypto_stream_gateway.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+## Mutual TLS based authentication
+
+In order to exchange messages with the Anbox Stream Agent through the [NATS](https://nats.io/) message queue, the Anbox Stream Gateway uses a CA certificated signed by a 4096 bit RSA key to ensure trust with the NATS server.
```
Comment (keirthana) [inline explanation/cryptography/crypto_stream_gateway.md:16]: As per Ubuntu style guide

```suggestion
To exchange messages with the Anbox Stream Agent through the [NATS](https://nats.io/) message queue, the Anbox Stream Gateway uses a CA certificate signed by a 4096 bit RSA key to ensure trust with the NATS server.
```

File: explanation/cryptography/crypto_stream_gateway.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+## Token based authentication
+
+Users can generate API tokens to authenticate with the HTTP API provided by the Anbox Stream Gateway. For the API tokens a scope-limited [Macaroon](http://theory.stanford.edu/~ataly/Papers/macaroons.pdf) is used. The token is signed with a [HMAC](https://www.okta.com/identity-101/hmac/) using SHA-256 (HS256) and a 64 byte secret key. The [`macaroon.New`](https://pkg.go.dev/gopkg.in/macaroon.v2@v2.1.0#New) method is used internally to generate the JWT token.
```
Comment (keirthana) [inline explanation/cryptography/crypto_stream_gateway.md:20]:
```suggestion
Users can generate API tokens to authenticate with the HTTP API provided by the Anbox Stream Gateway. For the API tokens, a scope-limited [Macaroon](http://theory.stanford.edu/~ataly/Papers/macaroons.pdf) is used. The token is signed with a [HMAC](https://www.okta.com/identity-101/hmac/) using SHA-256 (HS256) and a 64 byte secret key. The [`macaroon.New`](https://pkg.go.dev/gopkg.in/macaroon.v2@v2.1.0#New) method is used internally to generate the [JWT](https://jwt.io/) token.
```

**Review by keirthana — CHANGES_REQUESTED** [review]
Minor comments, otherwise looks good

File: explanation/cryptography/crypto_stream_gateway.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+All network endpoints exposed by the Anbox Stream Gateway are secured with TLS using an 4096 bit RSA key. The Anbox Stream Gateway strictly enforces TLS 1.3 or later and does not provide backward compatibility with older TLS versions.
+
+## Mutual TLS based authentication
+
+To exchange messages with the Anbox Stream Agent through the [NATS](https://nats.io/) message queue, the Anbox Stream Gateway uses a CA certificate signed by a 4096 bit RSA key to ensure trust with the NATS server.
+
+## Token based authentication
+
+Users can generate API tokens to authenticate with the HTTP API provided by the Anbox Stream Gateway. For the API tokens, a scope-limited [Macaroon](http://theory.stanford.edu/~ataly/Papers/macaroons.pdf) is used. The token is signed with a [HMAC](https://www.okta.com/identity-101/hmac/) using SHA-256 (HS256) and a 64 byte secret key. The [`macaroon.New`](https://pkg.go.dev/gopkg.in/macaroon.v2@v2.1.0#New) method is used internally to generate the [JWT](https://jwt.io/) token.
```
Comment (jat-canonical) [inline explanation/cryptography/crypto_stream_gateway.md:20]: > The [macaroon.New](https://pkg.go.dev/gopkg.in/macaroon.v2@v2.1.0#New) method is used internally to generate the [JWT](https://jwt.io/) token.

I wonder if this is necessary to include as this leaks internal implementation detail which might not be suitable for the nature of our product ?

**Review by jat-canonical — APPROVED** [review]
One questions otherwise LGTM

File: explanation/cryptography/crypto_stream_gateway.md (hunk 4 above)
Comment (morphis) [reply explanation/cryptography/crypto_stream_gateway.md:20]: That's fine. We reference the package we use already and drawing a conclusion to what method is used from it is straightforward.

**Pull request merged**